src/ui/public/vis/request_handlers/courier.js

```diff
@@ -5,8 +5,8 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
     name: 'courier',
     handler: function (vis, appState, uiState, searchSource) {
       if (appState) {
-        searchSource.set('filter', appState.filters);
-        if (!appState.linked) searchSource.set('query', appState.query);
+        searchSource.set('filter', _.union(searchSource.initialState.filter, appState.filters));
+        if (!appState.linked) searchSource.set('query', _.union([searchSource.initialState.query], [appState.query]));
```

Comment:
I don't think we should be merging this with a union of arrays. The problem is that we get other bugs:

- saving twice, will then save the visualization with an array in the searchSourceJSON. 
![image](https://user-images.githubusercontent.com/1833023/28141376-524d0efa-672a-11e7-8f45-79d7fe788df8.png)
- a side-effect is that the query-string of the visualization can no longer be restored correctly in the UI control
![image](https://user-images.githubusercontent.com/1833023/28141421-7bf5e4de-672a-11e7-98ec-403c8ebc779f.png)
- Another issue is, when the query is undefined, it gets merged into the object as null, creating malformed queries
- This in turn causes errors with the query, breaking both dashboards and visualizations.

![image](https://user-images.githubusercontent.com/1833023/28141574-29aa31e8-672b-11e7-9c5e-2d4a2315b7d0.png)